### PR TITLE
Add fetch depth capability

### PR DESCRIPTION
**Pull request by kerrmarin**
Hey,

This is a first stab at adding depth to the git-cmd (only for git, not for jgit). The tests could be better...

### CODE REVIEW

File: src/com/tw/go/plugin/jgit/JGitHelper.java [hunk 1]
```diff
@@ -293,6 +293,11 @@ public void fetch(String refSpec) {
     }
 
     @Override
+    public void fetch(String refSpec, Integer depth) {
+        fetch(refSpec); //Does not honour depth
```
Comment (ashwanthkumar) [inline src/com/tw/go/plugin/jgit/JGitHelper.java]: Shouldn't this throw something like NotImplemented exception?

File: test/com/tw/go/plugin/AbstractGitHelperTest.java [hunk 2]
```diff
@@ -259,6 +260,33 @@ public void shouldInitAddCommit() throws Exception {
         assertThat(revision.getModifiedFiles().get(0).getAction(), is("added"));
     }
 
+    @Test
+    public void shouldFetchWithDefaultDepth() {
+        GitConfig config = new GitConfig("https://github.com/mdaliejaz/samplerepo.git");
```
Comment (ashwanthkumar) [inline test/com/tw/go/plugin/AbstractGitHelperTest.java]: Using a github url is a little confusing here, can use something like 
```java
new GitConfig(testRepository.getAbsolutePath())
```

**Comment by ashwanthkumar** [comment]
This is a fantastic start @kerrmarin. I just a couple of minor comments, rest LGTM

File: test/com/tw/go/plugin/AbstractGitHelperTest.java (hunk 2 above)
Comment (kerrmarin) [inline test/com/tw/go/plugin/AbstractGitHelperTest.java]: I was going to do this -- but I saw the shallow clone tests were `@Ignored` and they didn't work, so I presumed the whole shallow-cloning a local repo didn't work? I'm happy to re-enable?

File: src/com/tw/go/plugin/jgit/JGitHelper.java (hunk 1 above)
Comment (kerrmarin) [inline src/com/tw/go/plugin/jgit/JGitHelper.java]: Sure, I'll change this -- I've never done this in Java so let me know if what I do is wrong.

**Comment by kerrmarin** [comment]
Hey, I've updated the tests and files. Let me know what you think.

**Comment by kerrmarin** [comment]
hey @ashwanthkumar, any feedback on this?

**Comment by ashwanthkumar** [comment]
@kerrmarin Just had a look. LGTM. I was thinking we could merge this along with the next release of the plugin. Meanwhile we could use a locally installed snapshot version for our testing purposes. Thoughts?

**Comment by kerrmarin** [comment]
Sounds good!

**Pull request closed without merging**